cav_msgs/msg/ManeuverParameters.msg

```diff
@@ -0,0 +1,38 @@
+# ManeuverParameters.msg
+#
+# Message defining shared parameters of Maneuvers in the CARMA platform
+#
+
+# Enum indicating the type of negotiation this maneuver is involved in
+uint8 neogition_type
+
+uint8 NO_NEGOTIATION = 0
+uint8 GENERAL_NEGOTIATION = 1
+uint8 PLATOONING = 2
+# Remaining enum values reserved for future use
+
+# The strategic plugin which planned this maneuver
+string planning_strategic_plugin
+
+# Optional Fields
+
+# Maneuver parameters presense vector
+
+uint16 presence_vector
+
+uint16 HAS_TACTICAL_PLUGIN = 1
+uint16 HAS_BOOL_META_DATA = 2
+uint16 HAS_INT_META_DATA = 4
+uint16 HAS_FLOAT_META_DATA = 8
+uint16 HAS_STRING_META_DATA = 16
+uint16 HAS_TIME_META_DATA = 32
+
+# The tactical plugin which planned this maneuver
+string planning_tactical_plugin
+
+# Optional meta data arrays which can be used to store additional information in the maneuver
```

Comment:
I'd personally favor a key/value based meta data system if that were plausible. A list of key/value pairs, keys always being strings, values being one of these types. Not sure how this message construction would look though.

Reply:
I would need to create a custom message type for each key value pair. There are existing Key/Value message types but they are all string/string